src/parser/warrior/protection/modules/features/BlockCheck.js

```diff
@@ -55,29 +64,48 @@ class BlockCheck extends Analyzer {
   on_toPlayer_damage(event) {
     // Physical
     if (event.ability.type === 1) {
-      if (this.selectedCombatant.hasBuff(SPELLS.SHIELD_BLOCK_BUFF.id) || (this.bolster && this.selectedCombatant.hasBuff(SPELLS.LAST_STAND.id))) {
-        this.physicalHitsWithShieldBlock += 1;
-        this.physicalDamageWithShieldBlock += event.amount + (event.absorbed || 0) + (event.overkill || 0);
-      } else {
-        this.physicalHitsWithoutShieldBlock += 1;
-        this.physicalDamageWithoutShieldBlock += event.amount + (event.absorbed || 0) + (event.overkill || 0);
-      }
+      event.prot = {
+        shieldBlock: this.selectedCombatant.hasBuff(SPELLS.SHIELD_BLOCK_BUFF.id),//ability to look what important buffs they had retro actively
+        bloster: (this.bolster && this.selectedCombatant.hasBuff(SPELLS.LAST_STAND.id)),
+      };
+      this.listOfEvents.push(event);
     }
   }
 
   on_fightend() {
+    const blockableSet = new Set();//this is master list of all BLOCKED events in the fight
+    blockableSet.add(1);//make it so if they never hit sb we still get data from the melees they take
+    this.shieldBlock.shieldBlocksDefensive.forEach(function(block){
+      block.eventName.forEach(function(blockedAbility){
```

Comment:
shouldn't this be `block.eventSpellId`?

Reply:
Yes. I added eventSpellId this pr and forgot to update